src/kayenta/metricStore/stackdriver/metricTypeSelector.tsx

```diff
@@ -8,60 +8,100 @@ import { AsyncRequestState } from 'kayenta/reducers/asyncRequest';
 import * as Creators from 'kayenta/actions/creators';
 import { DisableableReactSelect, DISABLE_EDIT_CONFIG } from 'kayenta/layout/disableable';
 
-interface IStackdriverMetricTypeSelectorDispatchProps {
+export interface IStackdriverMetricTypeSelectorDispatchProps {
   load: (filter: string) => void;
 }
 
-interface IStackdriverMetricTypeSelectorStateProps {
-  options: Option[];
+export interface IStackdriverMetricTypeSelectorStateProps {
+  descriptors: IStackdriverMetricDescriptor[];
   loading: boolean;
 }
 
-interface IStackdriverMetricTypeSelectorOwnProps {
+export interface IStackdriverMetricTypeSelectorOwnProps {
   value: string;
   onChange: (option: Option) => void;
 }
 
-const StackdriverMetricTypeSelector = ({
-  loading,
-  load,
-  options,
-  value,
-  onChange,
-}: IStackdriverMetricTypeSelectorDispatchProps &
+type IStackdriverMetricTypeProps = IStackdriverMetricTypeSelectorDispatchProps &
   IStackdriverMetricTypeSelectorStateProps &
-  IStackdriverMetricTypeSelectorOwnProps) => {
-  if (value && options.every(o => o.value !== value)) {
-    options = options.concat({ label: value, value });
-  }
+  IStackdriverMetricTypeSelectorOwnProps;
 
+const MetricField = ({
+  descriptor,
+  field,
+  label,
+}: {
+  descriptor: IStackdriverMetricDescriptor;
+  field: keyof IStackdriverMetricDescriptor;
+  label: string;
+}) => {
+  if (!descriptor || !descriptor[field]) {
+    return null;
+  }
   return (
-    <DisableableReactSelect
-      isLoading={loading}
-      options={options}
-      onChange={onChange}
-      value={value}
-      placeholder={'Enter at least three characters to search.'}
-      onInputChange={input => {
-        load(input);
-        return input;
-      }}
-      disabledStateKeys={[DISABLE_EDIT_CONFIG]}
-    />
+    <div>
+      <strong>{label}: </strong>
+      {descriptor[field]}
+    </div>
   );
 };
 
-const mapStateToProps = (state: ICanaryState, ownProps: IStackdriverMetricTypeSelectorOwnProps) => {
+export class StackdriverMetricTypeSelector extends React.Component<IStackdriverMetricTypeProps> {
+  public componentWillMount() {
```

Comment:
Might want to use `componentDidMount` instead to make future React version bumps easier (https://reactjs.org/docs/react-component.html#unsafe_componentwillmount)

Reply:
good point!